lib/shoehorn/handler.ex

```diff
@@ -1,21 +1,175 @@
 defmodule Shoehorn.Handler do
-  @callback init() :: any
+  @moduledoc """
+  A behaviour module for implementing handling of failing applications
 
-  @callback application_stopped(app :: atom, reason :: any, state :: any) ::
-              {:noreply, state :: any}
-              | {:stop, reason :: term, state :: any}
+  A Shoehorn.Handler is a module that knows how to respond to specific
+  applications going down. There are two types of failing applications.
+  The first is an application that fails to initialize and the second
+  is an application that stops while it is running.
 
-  defmacro __using__(_) do
+  ## Example
+
+  The Shoehorn.Handler behaviour requires developers to implement two
+  callbacks.
+
+  The `init` callback sets up any state the handler needs.
+
+  Ther `application_started` callback is called when an application
+  starts up.
+
+  The `application_exited` callback processes the incoming failure and
+  replies with the action that `Shoehorn` should take in case of
+  application failure.
+
+          defmodule Example.ShoehornHandler do
+            use Shoehorn.Handler
+
+            def init(_opts) do
+              {:ok, %{restart_counts: 0}}
+            end
+
+            def application_started(app, state) do
+              IO.inspect app
+              {:continue, state}
+            end
+
+            def application_exited(:non_esential_app, _reason,  state) do
```

Comment:
esential->essential. I've seen this typo at least twice, so it's probably worth a global find/replace.